fetch-in-react - Daniel Yount

TheDanYount opened the pull request.

TheDanYount committed:
Practiced frontend fetches w/ React

RobertGardner commented on fetch-in-react/my-app/src/UserDetails.tsx:
```diff
@@ -0,0 +1,50 @@
+        );
+        if (!result.ok)
+          throw new Error('could not fetch userData from clicked user');
+        const formattedResult = (await result.json()) as Promise<User>;
```
The `awaited` result is just `User`. `result.json()` is a Promise, but once it's awaited the promise gets removed.

RobertGardner commented on fetch-in-react/my-app/src/UserDetails.tsx:
```diff
@@ -0,0 +1,50 @@
+        if (!result.ok)
+          throw new Error('could not fetch userData from clicked user');
+        const formattedResult = (await result.json()) as Promise<User>;
+        setUser(await formattedResult);
```
So you don't need to `await` here.

RobertGardner commented on fetch-in-react/my-app/src/Users.tsx:
```diff
@@ -0,0 +1,72 @@
+          'https://jsonplaceholder.typicode.com/users'
+        );
+        if (!result.ok) throw new Error('initial users fetch failed');
+        const formattedResult = (await result.json()) as Promise<User[]>;
```
Same here.

RobertGardner reviewed (changes requested):
Minor fix.

TheDanYount committed:
Fixed sloppy code

Typed a json method response as a Promise and thus had to await; rewrote that in the two places I did it
